# Differential Expression Analysis {#diff-exp}

<iframe width="560" height="2" src="" frameborder="0" allowfullscreen></iframe>

//TODO insert video of performing differential expression analysis using Omics Central here

```{r, echo = FALSE, warning = FALSE, message = FALSE}
knitr::opts_chunk$set(cache = FALSE, echo = FALSE, warning = FALSE, message = FALSE, tidy = TRUE)

# setup
library(limma);  ## bioconductor
library(csSAM);
library(ggrepel);
source("helper_functions.R")

# load datasets
load(here::here("data", "biomonitoringBloodBiomarkersDatasets.Rdata"))
X.trainList$holter <- log2(X.trainList$holter)

## select dataset
X <- t(X.trainList$proteins)
avg <- rowMeans(X)
fc <- rowMeans(X[, hosp_3months == "Yes"]) - rowMeans(X[, hosp_3months == "No"])
```

## Methods

### Ordinary Least Squares

```{r}
ols_pvals <- apply(X, 1, function(i){
  fit <- lm(i~hosp_3months)
  coef(summary(fit))[2, "Pr(>|t|)"]
})
ols_fdr <- p.adjust(ols_pvals, "BH")
ols_fdr <- ols_fdr[order(ols_fdr)]
```

### LInear Models for MicroArrays and RNA-Seq

#### LIMMA

```{r}
design <- model.matrix(~hosp_3months)
fit <- eBayes(lmFit(X, design, method = "ls"))
top_limma <- topTable(fit, coef = "hosp_3monthsYes", n = nrow(fit), adjust.method = "BH")

limma_pvals <- top_limma$P.Value
limma_fdr <- top_limma$adj.P.Val
limma_fdr <- limma_fdr[order(limma_fdr)]
```

#### Robust LIMMA

```{r}
design <- model.matrix(~hosp_3months)
fit <- eBayes(lmFit(X, design, method = "robust"))
top_rlimma <- topTable(fit, coef = "hosp_3monthsYes", n = nrow(fit), adjust.method = "BH")

rlimma_pvals <- top_rlimma$P.Value
rlimma_fdr <- top_rlimma$adj.P.Val
rlimma_fdr <- rlimma_fdr[order(rlimma_fdr)]
```

#### LIMMA VOOM (adjusts for heteroscdasticity)

```{r}
v <- voom(2^X, design, plot=TRUE)
fit <- lmFit(v, design)
fit <- eBayes(fit)
top_vlimma <- topTable(fit, coef = "hosp_3monthsYes", n = nrow(fit), adjust.method = "BH")

vlimma_pvals <- top_vlimma$P.Value
vlimma_fdr <- top_vlimma$adj.P.Val
vlimma_fdr <- vlimma_fdr[order(vlimma_fdr)]
```

### Significance Analysis for Microarrays (SAM)

### cell-specific Analysis for Microarrays (csSAM)

```{r}
# from https://cran.r-project.org/web/packages/csSAM/csSAM.pdf
G <- X.trainList$proteins
cc <- X.trainList$cells
all(rownames(G) == rownames(cc))
## combine all lymphocytes into one since then number of cells is greater than the number of samples in one of the groups, and remove Eosinophils because most values are zeros
lyms <- c("B cells naive", "Plasma cells", "T cells CD8", "T cells CD4 naive", "T cells CD4 memory resting", "T cells CD4 memory activated", "T cells regulatory (Tregs)", "NK cells resting")
mono <- c("Monocytes", "Macrophages M2")
cc <- cbind(cc[, setdiff(colnames(cc), "Eosinophils")], "Lymphocytes"= rowSums(cc[, lyms, drop = FALSE]), "Monocytes"= rowSums(cc[, mono, drop = FALSE]))
cc <- cc[, setdiff(colnames(cc), c(lyms, mono))]


cellID <- colnames(cc)
numcell <- ncol(cc)
numgene <- ncol(G)
y <- hosp_3months
n <- table(y)

deconv <- list()
# run analysis
for (curset in levels(y)){
  deconv[[curset]]= csfit(cc[y==curset,], G[y==curset,])
}
rhat <- array(dim = c(numcell,numgene))
rhat[, ] <- csSAM(deconv[[1]]$ghat, deconv[[1]]$se,
n[1], deconv[[2]]$ghat, deconv[[2]]$se, n[2],
standardize=TRUE, medianCenter=TRUE, nonNeg=TRUE)
tt.sam <- runSAM2(G, y)

alternative <- c('two.sided', 'greater', 'less')
falseDiscovR <- lapply(alternative, function(i){
  falseDiscovR <- fdrCsSAM2(G,cc,y,n,numcell,numgene, rhat, 
    nperms = 200,standardize=TRUE,alternative=i, medianCenter=TRUE, nonNeg=TRUE)
  sigGene <- findSigGene(G, cc, y, rhat, falseDiscovR)
  colnames(sigGene) <- colnames(G)
  rownames(sigGene) <- colnames(cc)
  sigGene
})
names(falseDiscovR) <- alternative
# falseDiscovRSAM <- fdrSAM2(G, y, nperms=200, tt.sam , alternative = alternative)
# sam_fdr <- falseDiscovRSAM$sigGene.sam
# sam_fdr[sam_fdr > 1] <- 1
# sam_fdr <- sam_fdr[order(sam_fdr)]

```

## Visualizations

### Number of differentially expressed genes

```{r}
p <- ncol(G)
data.frame(n_sig = rep(1:p, 4),
           fdr = c(ols_fdr, limma_fdr, rlimma_fdr, vlimma_fdr),
           method = rep(c("OLS", "LIMMA", "rLIMMA", "vLIMMA"), each = p)) %>% 
  ggplot(aes(x = n_sig, y = fdr, color = method, group = method)) +
  geom_point() +
  geom_line() +
  scale_x_log10() +
  theme_classic() +
  ylab("False Discovery rate (FDR)") + 
  xlab("Number of significant genes")

```

### P-value histograms

```{r}
data.frame(pvals = c(ols_pvals, limma_pvals, rlimma_pvals, vlimma_pvals),
           method = rep(c("OLS", "LIMMA", "rLIMMA", "vLIMMA"), each = p)) %>% 
  ggplot(aes(x = pvals)) +
  geom_histogram(bins = 20) +
  theme_classic() +
  facet_wrap(~method) +
  geom_hline(yintercept = 5, linetype = "dashed", color = "gray") +
  xlab("p-values") + 
  ylab("Frequency")

```

### MA plot

```{r}
ma <- data.frame(fc = fc,
           pvals = -log10(ols_pvals[names(fc)]),
           sym = names(fc),
  fdr = ols_fdr[names(fc)])
maSubset <- ma %>% filter(fdr < 0.1)
ggplot(ma, aes(x = fc, y = pvals)) +
  geom_point() +
  geom_text_repel(data=maSubset, aes(x = fc, y = pvals, label = sym)) + 
  theme_classic() +
  geom_vline(xintercept = 0, linetype = "dashed", color = "gray") +
  xlab("Fold-change") + 
  ylab(expression('-log'[10]~'(p-value)'))
  
```

### csSAM

```{r}
ccsam_results <- data.frame(All = as.numeric(falseDiscovR$two.sided),
  Up = as.numeric(falseDiscovR$greater),
  Down = as.numeric(falseDiscovR$less))
ccsam_results$Feature <- rep(colnames(G), each = ncol(cc))
ccsam_results$Cell <- colnames(cc)

ccsam_results %>% 
  gather(Comparison, fdr, -c(Cell, Feature)) %>% 
  group_by(Comparison, Cell) %>% 
  arrange(fdr) %>% 
  mutate(nsig = dplyr::row_number()) %>% 
  ggplot(aes(x = nsig, y = fdr)) +
  geom_point() +
  scale_x_log10() +
  geom_line() +
  facet_grid(Comparison ~ Cell) +
  xlab("Number of significant genes") +
  ylab("False Discovery Rate") +
  theme_bw() +
  ylim(c(0,1))



```


