<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Biological Enrichment | Omics Central</title>
  <meta name="description" content="This book covers some commonly used to explore and analyze omics data." />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Biological Enrichment | Omics Central" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book covers some commonly used to explore and analyze omics data." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Biological Enrichment | Omics Central" />
  
  <meta name="twitter:description" content="This book covers some commonly used to explore and analyze omics data." />
  

<meta name="author" content="Amrit Singh" />


<meta name="date" content="2020-02-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-integration.html"/>
<link rel="next" href="lit.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Omics Central</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Rationale</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="data-types.html"><a href="data-types.html"><i class="fa fa-check"></i><b>3</b> Data-types</a><ul>
<li class="chapter" data-level="3.1" data-path="data-types.html"><a href="data-types.html#microarrays"><i class="fa fa-check"></i><b>3.1</b> Microarrays</a></li>
<li class="chapter" data-level="3.2" data-path="data-types.html"><a href="data-types.html#rna-sequencing"><i class="fa fa-check"></i><b>3.2</b> RNA sequencing</a></li>
<li class="chapter" data-level="3.3" data-path="data-types.html"><a href="data-types.html#nanostring"><i class="fa fa-check"></i><b>3.3</b> Nanostring</a></li>
<li class="chapter" data-level="3.4" data-path="data-types.html"><a href="data-types.html#biocrates"><i class="fa fa-check"></i><b>3.4</b> Biocrates</a></li>
<li class="chapter" data-level="3.5" data-path="data-types.html"><a href="data-types.html#multiple-reaction-monitoring"><i class="fa fa-check"></i><b>3.5</b> Multiple Reaction Monitoring</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="eda.html"><a href="eda.html"><i class="fa fa-check"></i><b>4</b> Exploratory Data Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="eda.html"><a href="eda.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="4.1.1" data-path="eda.html"><a href="eda.html#method"><i class="fa fa-check"></i><b>4.1.1</b> Method</a></li>
<li class="chapter" data-level="4.1.2" data-path="eda.html"><a href="eda.html#visualizations"><i class="fa fa-check"></i><b>4.1.2</b> Visualizations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="batch-correction.html"><a href="batch-correction.html"><i class="fa fa-check"></i><b>5</b> Batch Correction</a><ul>
<li class="chapter" data-level="5.1" data-path="batch-correction.html"><a href="batch-correction.html#combat"><i class="fa fa-check"></i><b>5.1</b> ComBat</a></li>
<li class="chapter" data-level="5.2" data-path="batch-correction.html"><a href="batch-correction.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>5.2</b> Surrogate Variable Analysis</a></li>
<li class="chapter" data-level="5.3" data-path="batch-correction.html"><a href="batch-correction.html#model-adjustment"><i class="fa fa-check"></i><b>5.3</b> Model adjustment</a></li>
<li class="chapter" data-level="5.4" data-path="batch-correction.html"><a href="batch-correction.html#references"><i class="fa fa-check"></i><b>5.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="diff-exp.html"><a href="diff-exp.html"><i class="fa fa-check"></i><b>6</b> Differential Expression Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="diff-exp.html"><a href="diff-exp.html#methods"><i class="fa fa-check"></i><b>6.1</b> Methods</a><ul>
<li class="chapter" data-level="6.1.1" data-path="diff-exp.html"><a href="diff-exp.html#ordinary-least-squares"><i class="fa fa-check"></i><b>6.1.1</b> Ordinary Least Squares</a></li>
<li class="chapter" data-level="6.1.2" data-path="diff-exp.html"><a href="diff-exp.html#linear-models-for-microarrays-and-rna-seq"><i class="fa fa-check"></i><b>6.1.2</b> LInear Models for MicroArrays and RNA-Seq</a></li>
<li class="chapter" data-level="6.1.3" data-path="diff-exp.html"><a href="diff-exp.html#significance-analysis-for-microarrays-sam"><i class="fa fa-check"></i><b>6.1.3</b> Significance Analysis for Microarrays (SAM)</a></li>
<li class="chapter" data-level="6.1.4" data-path="diff-exp.html"><a href="diff-exp.html#cell-specific-analysis-for-microarrays-cssam"><i class="fa fa-check"></i><b>6.1.4</b> cell-specific Analysis for Microarrays (csSAM)</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="diff-exp.html"><a href="diff-exp.html#visualizations-1"><i class="fa fa-check"></i><b>6.2</b> Visualizations</a><ul>
<li class="chapter" data-level="6.2.1" data-path="diff-exp.html"><a href="diff-exp.html#number-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>6.2.1</b> Number of differentially expressed genes</a></li>
<li class="chapter" data-level="6.2.2" data-path="diff-exp.html"><a href="diff-exp.html#p-value-histograms"><i class="fa fa-check"></i><b>6.2.2</b> P-value histograms</a></li>
<li class="chapter" data-level="6.2.3" data-path="diff-exp.html"><a href="diff-exp.html#ma-plot"><i class="fa fa-check"></i><b>6.2.3</b> MA plot</a></li>
<li class="chapter" data-level="6.2.4" data-path="diff-exp.html"><a href="diff-exp.html#cssam"><i class="fa fa-check"></i><b>6.2.4</b> csSAM</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>7</b> Network Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="network-analysis.html"><a href="network-analysis.html#dingo"><i class="fa fa-check"></i><b>7.1</b> DINGO</a></li>
<li class="chapter" data-level="7.2" data-path="network-analysis.html"><a href="network-analysis.html#wgcna"><i class="fa fa-check"></i><b>7.2</b> WGCNA</a></li>
<li class="chapter" data-level="7.3" data-path="network-analysis.html"><a href="network-analysis.html#panda"><i class="fa fa-check"></i><b>7.3</b> PANDA</a></li>
<li class="chapter" data-level="7.4" data-path="network-analysis.html"><a href="network-analysis.html#bionetstat"><i class="fa fa-check"></i><b>7.4</b> BioNetStat</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data Integration</a><ul>
<li class="chapter" data-level="8.1" data-path="data-integration.html"><a href="data-integration.html#supervised"><i class="fa fa-check"></i><b>8.1</b> Supervised</a><ul>
<li class="chapter" data-level="8.1.1" data-path="data-integration.html"><a href="data-integration.html#diablo-sgccda"><i class="fa fa-check"></i><b>8.1.1</b> DIABLO (SGCCDA)</a></li>
<li class="chapter" data-level="8.1.2" data-path="data-integration.html"><a href="data-integration.html#ensemble-of-glmnet-classifiers"><i class="fa fa-check"></i><b>8.1.2</b> Ensemble of glmnet classifiers</a></li>
<li class="chapter" data-level="8.1.3" data-path="data-integration.html"><a href="data-integration.html#diablo2-smb-plsda"><i class="fa fa-check"></i><b>8.1.3</b> DIABLO2 (sMB-PLSDA)</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="data-integration.html"><a href="data-integration.html#references-1"><i class="fa fa-check"></i><b>8.2</b> References</a></li>
<li class="chapter" data-level="8.3" data-path="data-integration.html"><a href="data-integration.html#unsupervised"><i class="fa fa-check"></i><b>8.3</b> Unsupervised</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-integration.html"><a href="data-integration.html#panda-1"><i class="fa fa-check"></i><b>8.3.1</b> PANDA</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-integration.html"><a href="data-integration.html#mofa"><i class="fa fa-check"></i><b>8.3.2</b> MOFA</a></li>
<li class="chapter" data-level="8.3.3" data-path="data-integration.html"><a href="data-integration.html#jive"><i class="fa fa-check"></i><b>8.3.3</b> JIVE</a></li>
<li class="chapter" data-level="8.3.4" data-path="data-integration.html"><a href="data-integration.html#snf"><i class="fa fa-check"></i><b>8.3.4</b> SNF</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="bio-enrichment.html"><a href="bio-enrichment.html"><i class="fa fa-check"></i><b>9</b> Biological Enrichment</a><ul>
<li class="chapter" data-level="9.1" data-path="bio-enrichment.html"><a href="bio-enrichment.html#enrichr"><i class="fa fa-check"></i><b>9.1</b> Enrichr</a></li>
<li class="chapter" data-level="9.2" data-path="bio-enrichment.html"><a href="bio-enrichment.html#sear"><i class="fa fa-check"></i><b>9.2</b> SEAR</a><ul>
<li class="chapter" data-level="9.2.1" data-path="bio-enrichment.html"><a href="bio-enrichment.html#hypergeometric-tests"><i class="fa fa-check"></i><b>9.2.1</b> hypergeometric tests</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="bio-enrichment.html"><a href="bio-enrichment.html#camera"><i class="fa fa-check"></i><b>9.3</b> CAMERA</a></li>
<li class="chapter" data-level="9.4" data-path="bio-enrichment.html"><a href="bio-enrichment.html#network-based-gene-set-analysis"><i class="fa fa-check"></i><b>9.4</b> Network-based Gene Set Analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="lit.html"><a href="lit.html"><i class="fa fa-check"></i><b>10</b> Literature Mining</a></li>
<li class="chapter" data-level="" data-path="references-3.html"><a href="references-3.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics Central</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bio-enrichment" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Biological Enrichment</h1>
<div id="enrichr" class="section level2">
<h2><span class="header-section-number">9.1</span> Enrichr</h2>
</div>
<div id="sear" class="section level2">
<h2><span class="header-section-number">9.2</span> SEAR</h2>
<div id="hypergeometric-tests" class="section level3">
<h3><span class="header-section-number">9.2.1</span> hypergeometric tests</h3>
<div id="hypergeometric-probabilities" class="section level4">
<h4><span class="header-section-number">9.2.1.1</span> hypergeometric probabilities</h4>
<ul>
<li>The sample space consists of a total of n genes, out of which m genes belong to Pathway A. Select k genes at random (without replacement). What is the probability that i of the selected genes belong to Pathway A.</li>
<li>parameters include:</li>
<li>n: total number of genes observed</li>
<li>m: number of genes in Pathway A</li>
<li>k: genes selected at random</li>
<li>i: number of selected genes that belong to Pathway A</li>
</ul>
<p>Size of sample space (<span class="math inline">\(\Omega \)</span>) = <span class="math inline">\(n \choose k \)</span>: all ways to draw k genes from n genes</p>
<p>Event of interest: # of ways to get i genes from Pathway A after drawing k genes = (# of ways to select i genes from Pathway A from a total of m genes in Pathway A, <span class="math inline">\(m \choose i \)</span>) x (# of ways to get k-i from the remaining n-m genes not in Pathway A, <span class="math inline">\(n-m \choose k-i \)</span>)</p>
</div>
<div id="asthma-case-study" class="section level4">
<h4><span class="header-section-number">9.2.1.2</span> Asthma case study</h4>
<div id="step-1-number-of-genes-measured" class="section level5">
<h5><span class="header-section-number">9.2.1.2.1</span> Step 1: Number of genes measured</h5>
<p>5443 gene transcripts were profiled in 28 blood samples from asthmatic individual undergoing allergen inhalation challenge.</p>
</div>
<div id="step-2-number-of-genes-observed-in-the-db" class="section level5">
<h5><span class="header-section-number">9.2.1.2.2</span> Step 2: Number of genes observed in the DB</h5>
<p>The <a href="https://github.com/singha53/omicsCentralDatasets/blob/master/inst/extdata/dataCleaning/pathwayDB/pathways.md">KEGG database</a> consisted of 7802 genes.</p>
</div>
<div id="step-3-overlap-between-gene-dataset-and-pathway-db" class="section level5">
<h5><span class="header-section-number">9.2.1.2.3</span> Step 3: Overlap between gene dataset and pathway DB</h5>
<p>There were 5357 genes that overlapped between the genes measured and those observed in the DB.</p>
</div>
<div id="step-4-keep-common-genes-in-gene-expression-dataset-and-kegg-db." class="section level5">
<h5><span class="header-section-number">9.2.1.2.4</span> Step 4: Keep common genes in gene expression dataset and KEGG DB.</h5>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
</div>
<div id="number-of-differentially-expressed-genes-between-pre-and-post-allergen-inhalation-challenge" class="section level4">
<h4><span class="header-section-number">9.2.1.3</span> Number of differentially expressed genes between pre and post allergen inhalation challenge</h4>
</div>
<div id="p-value-of-randomly-selecting-50-genes-from-all-observed-genes" class="section level4">
<h4><span class="header-section-number">9.2.1.4</span> P-value of randomly selecting 50 genes from all observed genes</h4>
<pre><code>##       Pathways    Genes              DB
## 4448    Asthma     IL10 KEGG_2019_Human
## 5064    Asthma     IL13 KEGG_2019_Human
## 5680    Asthma     PRG2 KEGG_2019_Human
## 5988    Asthma   RNASE3 KEGG_2019_Human
## 6292    Asthma      IL3 KEGG_2019_Human
## 6594    Asthma      IL5 KEGG_2019_Human
## 6896    Asthma      IL4 KEGG_2019_Human
## 7196    Asthma      IL9 KEGG_2019_Human
## 7793    Asthma HLA-DQB1 KEGG_2019_Human
## 8089    Asthma HLA-DPB1 KEGG_2019_Human
## 8385    Asthma   CD40LG KEGG_2019_Human
## 8973    Asthma    MS4A2 KEGG_2019_Human
## 9848    Asthma     CD40 KEGG_2019_Human
## 10136   Asthma    CCL11 KEGG_2019_Human
## 10421   Asthma      TNF KEGG_2019_Human
## 10704   Asthma      EPX KEGG_2019_Human
## 10987   Asthma  HLA-DMB KEGG_2019_Human
## 11268   Asthma HLA-DPA1 KEGG_2019_Human
## 11543   Asthma   FCER1A KEGG_2019_Human
## 11817   Asthma   FCER1G KEGG_2019_Human
## 12089   Asthma  HLA-DOB KEGG_2019_Human
## 12358   Asthma  HLA-DMA KEGG_2019_Human
## 12888   Asthma  HLA-DOA KEGG_2019_Human
## 13151   Asthma HLA-DQA1 KEGG_2019_Human</code></pre>
</div>
<div id="references-2" class="section level4">
<h4><span class="header-section-number">9.2.1.5</span> References</h4>
<ol style="list-style-type: decimal">
<li><a href="https://www.edx.org/course/probability-the-science-of-uncertainty-and-data">Probability - The Science of Uncertainty and Data</a></li>
<li>Falcon S., Gentleman R. (2008) Hypergeometric Testing Used for Gene Set Enrichment Analysis. In: Bioconductor Case Studies. Use R!. Springer, New York, NY</li>
</ol>
</div>
</div>
</div>
<div id="camera" class="section level2">
<h2><span class="header-section-number">9.3</span> CAMERA</h2>
</div>
<div id="network-based-gene-set-analysis" class="section level2">
<h2><span class="header-section-number">9.4</span> Network-based Gene Set Analysis</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-integration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="lit.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/08-bio-enrichment.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
