<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Exploratory Data Analysis | Omics Central</title>
  <meta name="description" content="This book covers some commonly used to explore and analyze omics data." />
  <meta name="generator" content="bookdown 0.17.2 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Exploratory Data Analysis | Omics Central" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book covers some commonly used to explore and analyze omics data." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Exploratory Data Analysis | Omics Central" />
  
  <meta name="twitter:description" content="This book covers some commonly used to explore and analyze omics data." />
  

<meta name="author" content="Amrit Singh" />


<meta name="date" content="2020-01-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-types.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Omics Central</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Rationale</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="data-types.html"><a href="data-types.html"><i class="fa fa-check"></i><b>3</b> Data-types</a><ul>
<li class="chapter" data-level="3.1" data-path="data-types.html"><a href="data-types.html#microarrays"><i class="fa fa-check"></i><b>3.1</b> Microarrays</a></li>
<li class="chapter" data-level="3.2" data-path="data-types.html"><a href="data-types.html#rna-sequencing"><i class="fa fa-check"></i><b>3.2</b> RNA sequencing</a></li>
<li class="chapter" data-level="3.3" data-path="data-types.html"><a href="data-types.html#nanostring"><i class="fa fa-check"></i><b>3.3</b> Nanostring</a></li>
<li class="chapter" data-level="3.4" data-path="data-types.html"><a href="data-types.html#biocrates"><i class="fa fa-check"></i><b>3.4</b> Biocrates</a></li>
<li class="chapter" data-level="3.5" data-path="data-types.html"><a href="data-types.html#multiple-reaction-monitoring"><i class="fa fa-check"></i><b>3.5</b> Multiple Reaction Monitoring</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="eda.html"><a href="eda.html"><i class="fa fa-check"></i><b>4</b> Exploratory Data Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="eda.html"><a href="eda.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="4.1.1" data-path="eda.html"><a href="eda.html#method"><i class="fa fa-check"></i><b>4.1.1</b> Method</a></li>
<li class="chapter" data-level="4.1.2" data-path="eda.html"><a href="eda.html#visualizations"><i class="fa fa-check"></i><b>4.1.2</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="eda.html"><a href="eda.html#k-means"><i class="fa fa-check"></i><b>4.2</b> K-Means</a></li>
<li class="chapter" data-level="4.3" data-path="eda.html"><a href="eda.html#hierarchical-clustering"><i class="fa fa-check"></i><b>4.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="4.4" data-path="eda.html"><a href="eda.html#sample-plots"><i class="fa fa-check"></i><b>4.4</b> Sample plots</a><ul>
<li class="chapter" data-level="4.4.1" data-path="eda.html"><a href="eda.html#sample-correlation-heatmap"><i class="fa fa-check"></i><b>4.4.1</b> Sample correlation heatmap</a></li>
<li class="chapter" data-level="4.4.2" data-path="eda.html"><a href="eda.html#sample-histograms"><i class="fa fa-check"></i><b>4.4.2</b> Sample histograms</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="eda.html"><a href="eda.html#variable-plots"><i class="fa fa-check"></i><b>4.5</b> Variable plots</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>5</b> References</a></li>
<li class="chapter" data-level="6" data-path="batch-correction.html"><a href="batch-correction.html"><i class="fa fa-check"></i><b>6</b> Batch Correction</a><ul>
<li class="chapter" data-level="6.1" data-path="batch-correction.html"><a href="batch-correction.html#combat"><i class="fa fa-check"></i><b>6.1</b> ComBat</a></li>
<li class="chapter" data-level="6.2" data-path="batch-correction.html"><a href="batch-correction.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>6.2</b> Surrogate Variable Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="batch-correction.html"><a href="batch-correction.html#model-adjustment"><i class="fa fa-check"></i><b>6.3</b> Model adjustment</a></li>
<li class="chapter" data-level="6.4" data-path="batch-correction.html"><a href="batch-correction.html#references-1"><i class="fa fa-check"></i><b>6.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diff-exp.html"><a href="diff-exp.html"><i class="fa fa-check"></i><b>7</b> Differential Expression Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="diff-exp.html"><a href="diff-exp.html#ordinary-least-squares"><i class="fa fa-check"></i><b>7.1</b> Ordinary Least Squares</a></li>
<li class="chapter" data-level="7.2" data-path="diff-exp.html"><a href="diff-exp.html#linear-models-for-microarrays-and-rna-seq"><i class="fa fa-check"></i><b>7.2</b> LInear Models for MicroArrays and RNA-Seq</a><ul>
<li class="chapter" data-level="7.2.1" data-path="diff-exp.html"><a href="diff-exp.html#robust-limma"><i class="fa fa-check"></i><b>7.2.1</b> Robust LIMMA</a></li>
<li class="chapter" data-level="7.2.2" data-path="diff-exp.html"><a href="diff-exp.html#limma-voom"><i class="fa fa-check"></i><b>7.2.2</b> LIMMA VOOM</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diff-exp.html"><a href="diff-exp.html#significance-analysis-for-microarrays-sam"><i class="fa fa-check"></i><b>7.3</b> Significance Analysis for Microarrays (SAM)</a></li>
<li class="chapter" data-level="7.4" data-path="diff-exp.html"><a href="diff-exp.html#cell-specific-analysis-for-microarrays-cssam"><i class="fa fa-check"></i><b>7.4</b> cell-specific Analysis for Microarrays (csSAM)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>8</b> Network Analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="network-analysis.html"><a href="network-analysis.html#dingo"><i class="fa fa-check"></i><b>8.1</b> DINGO</a></li>
<li class="chapter" data-level="8.2" data-path="network-analysis.html"><a href="network-analysis.html#wgcna"><i class="fa fa-check"></i><b>8.2</b> WGCNA</a></li>
<li class="chapter" data-level="8.3" data-path="network-analysis.html"><a href="network-analysis.html#panda"><i class="fa fa-check"></i><b>8.3</b> PANDA</a></li>
<li class="chapter" data-level="8.4" data-path="network-analysis.html"><a href="network-analysis.html#bionetstat"><i class="fa fa-check"></i><b>8.4</b> BioNetStat</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>9</b> Data Integration</a><ul>
<li class="chapter" data-level="9.1" data-path="data-integration.html"><a href="data-integration.html#supervised"><i class="fa fa-check"></i><b>9.1</b> Supervised</a><ul>
<li class="chapter" data-level="9.1.1" data-path="data-integration.html"><a href="data-integration.html#diablo-sgccda"><i class="fa fa-check"></i><b>9.1.1</b> DIABLO (SGCCDA)</a></li>
<li class="chapter" data-level="9.1.2" data-path="data-integration.html"><a href="data-integration.html#ensemble-of-glmnet-classifiers"><i class="fa fa-check"></i><b>9.1.2</b> Ensemble of glmnet classifiers</a></li>
<li class="chapter" data-level="9.1.3" data-path="data-integration.html"><a href="data-integration.html#diablo2-smb-plsda"><i class="fa fa-check"></i><b>9.1.3</b> DIABLO2 (sMB-PLSDA)</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="data-integration.html"><a href="data-integration.html#unsupervised"><i class="fa fa-check"></i><b>9.2</b> Unsupervised</a><ul>
<li class="chapter" data-level="9.2.1" data-path="data-integration.html"><a href="data-integration.html#panda-1"><i class="fa fa-check"></i><b>9.2.1</b> PANDA</a></li>
<li class="chapter" data-level="9.2.2" data-path="data-integration.html"><a href="data-integration.html#mofa"><i class="fa fa-check"></i><b>9.2.2</b> MOFA</a></li>
<li class="chapter" data-level="9.2.3" data-path="data-integration.html"><a href="data-integration.html#jive"><i class="fa fa-check"></i><b>9.2.3</b> JIVE</a></li>
<li class="chapter" data-level="9.2.4" data-path="data-integration.html"><a href="data-integration.html#snf"><i class="fa fa-check"></i><b>9.2.4</b> SNF</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="bio-enrichment.html"><a href="bio-enrichment.html"><i class="fa fa-check"></i><b>10</b> Biological Enrichment</a><ul>
<li class="chapter" data-level="10.1" data-path="bio-enrichment.html"><a href="bio-enrichment.html#enrichr"><i class="fa fa-check"></i><b>10.1</b> Enrichr</a></li>
<li class="chapter" data-level="10.2" data-path="bio-enrichment.html"><a href="bio-enrichment.html#sear"><i class="fa fa-check"></i><b>10.2</b> SEAR</a></li>
<li class="chapter" data-level="10.3" data-path="bio-enrichment.html"><a href="bio-enrichment.html#camera"><i class="fa fa-check"></i><b>10.3</b> CAMERA</a></li>
<li class="chapter" data-level="10.4" data-path="bio-enrichment.html"><a href="bio-enrichment.html#network-based-gene-set-analysis"><i class="fa fa-check"></i><b>10.4</b> Network-based Gene Set Analysis</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="lit.html"><a href="lit.html"><i class="fa fa-check"></i><b>11</b> Literature Mining</a></li>
<li class="chapter" data-level="" data-path="references-2.html"><a href="references-2.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics Central</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="eda" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Exploratory Data Analysis</h1>
<div id="principal-component-analysis" class="section level2">
<h2><span class="header-section-number">4.1</span> Principal Component Analysis</h2>
<iframe width="560" height="2" src frameborder="0" allowfullscreen>
</iframe>
<p>//TODO insert video of EDA using Omics Central here</p>
<div id="method" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Method</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## select dataset
X &lt;-<span class="st"> </span>X.trainList<span class="op">$</span>clinical

## constants
n &lt;-<span class="st"> </span><span class="kw">length</span>(hosp_3months) <span class="co"># number of samples</span>
k &lt;-<span class="st"> </span><span class="dv">5</span> <span class="co"># number of PCs</span>
p &lt;-<span class="st"> </span><span class="kw">ncol</span>(X) <span class="co"># number of variables</span>

## run Principal Component Analysis
pca &lt;-<span class="st"> </span><span class="kw">prcomp</span>(X, <span class="dt">scale. =</span> <span class="ot">TRUE</span>)</code></pre></div>
<blockquote>
<p>prcomp() performs Principal Component Analysis (PCA) on the clinical dataset X (n observations (samples) x p variables).</p>
</blockquote>
<div id="what-is-pca" class="section level4">
<h4><span class="header-section-number">4.1.1.1</span> What is PCA?</h4>
<ul>
<li>method to turn a dataset with correlated variables into another dataset with linearly uncorrelated variables called principal components (PCs).</li>
</ul>
</div>
<div id="why-is-pca-useful" class="section level4">
<h4><span class="header-section-number">4.1.1.2</span> Why is PCA useful?</h4>
<ul>
<li>The first few PCs capture most of the variability in the data.</li>
<li>PCA can be used to visualize clustering patterns (samples or variables) in the data, determine relationships between samples (see Principal Component plot), between variables (see Correlation circle), between samples and variables (see Biplot).</li>
<li>PCA is also useful in determining the influece of covariates, both techincal (<em>e.g.</em> batch effects) or biological (<em>e.g.</em> sex).</li>
</ul>
</div>
<div id="what-is-a-principal-component-pc" class="section level4">
<h4><span class="header-section-number">4.1.1.3</span> What is a principal component (PC)?</h4>
<ul>
<li>a PC is a weighted average of the original predictors, <strong>PC</strong><sub><em>i</em></sub> = <strong>Xv</strong><sub><em>i</em></sub>, where <strong>X</strong> is a centered matrix and <em>i=1,…,n</em>.</li>
</ul>
</div>
<div id="what-do-the-vector-of-weights-vi-do" class="section level4">
<h4><span class="header-section-number">4.1.1.4</span> What do the vector of weights <strong>v</strong><sub><em>i</em></sub> do?</h4>
<ul>
<li><em>v<sub>i</sub></em> maximizes the variance; <strong>X<sup>T</sup>X</strong> and are called eigenvectors, weights or loadings.</li>
</ul>
</div>
<div id="how-do-i-compute-the-vector-of-weights-vi" class="section level4">
<h4><span class="header-section-number">4.1.1.5</span> How do I compute the vector of weights, <em>v<sub>i</sub></em>?</h4>
<ul>
<li>apply a factorization method called singular value decomposition (SVD). SVD decomposes a matrix X into a product of 3 matrices, <strong>UDV<sup>T</sup></strong>; <strong>X</strong><sub><em>np</em></sub> = <strong>U</strong><sub><em>nxp</em></sub> x <strong>D</strong>~<em>pxp~</em> x <strong>V<sup>T</sup></strong><sub><em>pp</em></sub> or <strong>X<sup>T</sup>X</strong> = <strong>VD<sup>2</sup>V<sup>T</sup></strong>.</li>
<li>The columns of <strong>V</strong> are the weights/loadings for each principal component.</li>
<li><strong>D</strong> is a diagnoal matrix where entry <strong>D</strong><sub><em>i,i</em></sub> is the standard deviation of the <em>ith</em> principal component (PC).</li>
<li>Only the first <em>k</em> PCs are needed to capture the majority of the variation in the high dimensional dataset (<em>n</em> &lt;&lt; <em>p</em> and <em>k</em> &lt;&lt; <em>p</em>); <strong>X</strong><sub><em>nk</em></sub> = <strong>U</strong><sub><em>nxk</em></sub> x <strong>D</strong><sub><em>pxk</em></sub> x <strong>V<sup>T</sup></strong><sub><em>nk</em></sub> such that <strong>X</strong><sub><em>nk</em></sub> <span class="math inline">\(\approx\)</span> <strong>X</strong><sub><em>np</em></sub>.</li>
</ul>
</div>
<div id="why-scale-the-data-before-applying-pca" class="section level4">
<h4><span class="header-section-number">4.1.1.6</span> Why scale the data before applying PCA?</h4>
<ul>
<li>The clinical variables are on different unit scales (<em>e.g.</em> Age (years) <em>vs.</em> Ejection fraction (%)). Scaling makes the mean of each variable zero and the standard deviation one.</li>
</ul>
<p>References<br />
1. page 64-66 from ESL: <a href="https://web.stanford.edu/~hastie/ElemStatLearn/printings/ESLII_print10.pdf" class="uri">https://web.stanford.edu/~hastie/ElemStatLearn/printings/ESLII_print10.pdf</a><br />
2. Wikipedia: <a href="https://en.wikipedia.org/wiki/Principal_component_analysis" class="uri">https://en.wikipedia.org/wiki/Principal_component_analysis</a></p>
</div>
</div>
<div id="visualizations" class="section level3">
<h3><span class="header-section-number">4.1.2</span> Visualizations</h3>
<div id="scree-plot" class="section level4">
<h4><span class="header-section-number">4.1.2.1</span> Scree plot</h4>
<ul>
<li>determine the proportion of variation explained by each principal component.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">propVar &lt;-<span class="st"> </span>(<span class="dv">100</span> <span class="op">*</span><span class="st"> </span>pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span><span class="op">/</span><span class="kw">sum</span>(pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span>))[<span class="dv">1</span><span class="op">:</span>k]
<span class="kw">data.frame</span>(<span class="dt">var =</span> propVar,
  <span class="dt">comp =</span> <span class="kw">factor</span>(<span class="kw">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="dv">1</span><span class="op">:</span>k), <span class="kw">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="dv">1</span><span class="op">:</span>k))) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(comp, var)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Principal Component&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion of variance explained (%)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Scree plot&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_classic</span>()</code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<blockquote>
<p>The barplot depicts the proportion of variation that is captured by the first five PCs; the first PC captures ~15.3% of the variability in the dataset consisting of 26 variables.</p>
</blockquote>
</div>
<div id="component-plot" class="section level4">
<h4><span class="header-section-number">4.1.2.2</span> Component plot</h4>
<ul>
<li>visualize the clustering of the samples and identify any clustering with respect to covariates of interest.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## plot pca$x
<span class="kw">fviz_pca_ind</span>(pca, <span class="dt">label=</span><span class="st">&quot;none&quot;</span>, <span class="dt">habillage=</span>hosp_3months,
  <span class="dt">addEllipses=</span><span class="ot">TRUE</span>, <span class="dt">ellipse.level=</span><span class="fl">0.80</span>)</code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<blockquote>
<p>The scatter plot above is a 2D depiction of a 26 (# of clinical variables) dimensional dataset. PC1 and PC2 together capture 25.4% of the variability in the clinical dataset. Some separation between the groups of interest can be observed.</p>
</blockquote>
</div>
<div id="correlation-circle" class="section level4">
<h4><span class="header-section-number">4.1.2.3</span> Correlation Circle</h4>
<ul>
<li><p>determine relationship between variables (based on the correlation between each variable and PCs).</p></li>
<li>the angle (<span class="math inline">\(\theta\)</span>) between two vectors determines the correlation between the two variables:<br />
</li>
<li><span class="math inline">\(\theta\)</span>=0: postive correlation (corr=1)<br />
</li>
<li>0&lt;<span class="math inline">\(\theta\)</span>&lt;90: postive correlation<br />
</li>
<li><span class="math inline">\(\theta\)</span>=90: zero correlation<br />
</li>
<li>90&lt;<span class="math inline">\(\theta\)</span>&lt;180: negative correlation<br />
</li>
<li><p><span class="math inline">\(\theta\)</span>=180: negative correlation (corr=-1)</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fviz_pca_var</span>(pca, <span class="dt">repel =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="correlation-circle-with-a-cut-off" class="section level4">
<h4><span class="header-section-number">4.1.2.4</span> Correlation Circle (with a cut-off)</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># compute correlation between variables and PCs</span>
corr &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">cor</span>(pca<span class="op">$</span>x[, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], X))

<span class="co"># correlation circle</span>
cor_cutoff &lt;-<span class="st"> </span><span class="fl">0.5</span>
corr <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">label =</span> <span class="kw">rownames</span>(.)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(<span class="kw">abs</span>(PC1) <span class="op">&gt;</span><span class="st"> </span>cor_cutoff <span class="op">|</span><span class="st"> </span><span class="kw">abs</span>(PC2) <span class="op">&gt;</span><span class="st"> </span>cor_cutoff) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_circle</span>(<span class="kw">aes</span>(<span class="dt">x0 =</span> <span class="dv">0</span>, <span class="dt">y0 =</span> <span class="dv">0</span>, <span class="dt">r =</span> <span class="dv">1</span>), <span class="dt">color =</span> <span class="st">&quot;gray&quot;</span>, <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_circle</span>(<span class="kw">aes</span>(<span class="dt">x0 =</span> <span class="dv">0</span>, <span class="dt">y0 =</span> <span class="dv">0</span>, <span class="dt">r =</span> cor_cutoff), <span class="dt">color =</span> <span class="st">&quot;gray&quot;</span>, <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">xend =</span> PC1, <span class="dt">yend =</span> PC2), <span class="dt">color =</span> <span class="st">&quot;gray&quot;</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.03</span>, <span class="st">&quot;npc&quot;</span>))) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_text_repel</span>(<span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">label =</span> label)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Correlation with PC1&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Correlation with PC2&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Correlation Circle&quot;</span>)</code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<blockquote>
<p>The above plot only displays the variables if they have a correlation greater than 0.5 with either PC1 or PC2. Ischemia and Statins are positively correlated suggesting that patients with ischemia are likely to be on statins. BNP (Brain Natriuretic Peptide) is positively correlated with age and negatively correlated with Heart Rate.</p>
</blockquote>
<p>References<br />
1. Figure 1 from BioData Mining volume 5, Article number: 19 (2012)<br />
2. plotVar(): mixOmics R-library 3. fviz_pca_var(): factoextra R-library</p>
</div>
<div id="biplot" class="section level4">
<h4><span class="header-section-number">4.1.2.5</span> Biplot</h4>
<ul>
<li>superimpose the principal components with loadings vectors.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fviz_pca_biplot</span>(pca,
      <span class="dt">geom =</span> <span class="st">&quot;point&quot;</span>,
      <span class="dt">col.ind =</span> <span class="st">&quot;cos2&quot;</span>,
      <span class="dt">habillage=</span>hosp_3months,
      <span class="dt">repel =</span> <span class="ot">TRUE</span>,    
      <span class="dt">addEllipses=</span><span class="ot">TRUE</span>, <span class="dt">ellipse.level=</span><span class="fl">0.80</span>) <span class="op">+</span>
<span class="st">      </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<blockquote>
<p>Each arrow can be thought of as an axis. For example, BNP points to the left which means that patients on the left (PC1 &lt; 0) have lower BNP levels than patients on the right (PC1 &gt; 0). Patients at the center (PC=1) have an average BNP level. Note that this aligns well with the hospitalization status; <em>ie.</em> patients on the left are more likely to be hospitalized as compared to patients on the right.</p>
</blockquote>
<p>References<br />
1. ggbiplot(): <a href="https://github.com/vqv/ggbiplot" class="uri">https://github.com/vqv/ggbiplot</a><br />
2. Biplot: <a href="https://stackoverflow.com/questions/6578355/plotting-pca-biplot-with-ggplot2" class="uri">https://stackoverflow.com/questions/6578355/plotting-pca-biplot-with-ggplot2</a><br />
3. biplot(): K. R. Gabriel (1971). The biplot graphical display of matrices with application to principal component analysis. Biometrika, 58, 453–467. doi: 10.2307/2334381.<br />
4. fviz_pca_biplot(): <a href="http://www.sthda.com/english/wiki/fviz-pca-quick-principal-component-analysis-data-visualization-r-software-and-data-mining" class="uri">http://www.sthda.com/english/wiki/fviz-pca-quick-principal-component-analysis-data-visualization-r-software-and-data-mining</a></p>
</div>
</div>
</div>
<div id="k-means" class="section level2">
<h2><span class="header-section-number">4.2</span> K-Means</h2>
</div>
<div id="hierarchical-clustering" class="section level2">
<h2><span class="header-section-number">4.3</span> Hierarchical clustering</h2>
</div>
<div id="sample-plots" class="section level2">
<h2><span class="header-section-number">4.4</span> Sample plots</h2>
<div id="sample-correlation-heatmap" class="section level3">
<h3><span class="header-section-number">4.4.1</span> Sample correlation heatmap</h3>
</div>
<div id="sample-histograms" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Sample histograms</h3>
</div>
</div>
<div id="variable-plots" class="section level2">
<h2><span class="header-section-number">4.5</span> Variable plots</h2>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-types.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-eda.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
